src/hooks/useAnimatedPageScrollHandler.ts

```diff
@@ -0,0 +1,186 @@
+import { MutableRefObject, RefObject, useEffect, useRef } from 'react';
```

Comment:
This code is mostly copied from https://github.com/software-mansion/react-native-reanimated/blob/master/src/reanimated2/hook/useAnimatedScrollHandler.ts. 


We will be able to remove some of those lines after stable 2.3.0 release